juju/framework.py

```diff
@@ -226,9 +226,13 @@ def events(self):
         """Return a mapping of event_kinds to bound_events for all available events.
         """
         events_map = {}
-        for event_kind in dir(self):
-            bound_event = getattr(self, event_kind)
-            if isinstance(bound_event, BoundEvent):
+        for event_kind in dir(type(self)):
+            # filter based on class attr value in case there are properties
+            # defined on subclasses to access sets of events, which could
+            # cause infinite recursion
```

Comment:
Another soft-agreement we had last week is to make sure comments are proper sentences, with capitalization, punctuation, etc. This is analogous to the remarks made above, and make them more pleasant to read by passers by.